tests/publiccloud/sles4sap.pm

```diff
@@ -0,0 +1,58 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test public cloud SLES4SAP images
+#
+# Maintainer: Loic Devulder <ldevulder@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use Mojo::File 'path';
+use Mojo::JSON;
+
+sub upload_ha_sap_logs {
+    my ($self, $instance) = @_;
+    my @logfiles = qw(provisioning.log salt-deployment.log salt-formula.log salt-pre-installation.log);
+
+    # Upload logs from public cloud VM
+    $instance->run_ssh_command(cmd => 'sudo chmod o+r /tmp/*.log');
+    foreach my $file (@logfiles) {
+        $instance->upload_log("/tmp/$file", log_name => $instance->{instance_id});
+    }
+}
+
+sub run {
+    my ($self) = @_;
+
+    $self->select_serial_terminal;
+
+    my $provider = $self->provider_factory();
+    foreach my $instance ($provider->create_instance(check_connectivity => 1)) {
```

Comment:
So the trick is, that you basically do all test execution in your terraform files, right?

Reply:
Can you plz point me to it. I tried to follow but I was lost in salts....

Reply:
Yes all is done in the terraform files. But I can't explain you all the details... I first create a GCP shell based version and now my colleagues and the SHAP dev team are using Salt, and I also have some issue to understand all... I'm improving this but that not so easy :-)

Basically what is done is: VM deployment with terraform, Salt recipes is then used to configure the VMs, install SAP HANA and create an HA cluster with HANA. Then for QA we will add fencing test in openQA (later) and then the VMs are destroyed. All logs are also uploaded to check after execution if needed.
The deployment is done in foreground mode (background can be used also, but not useful for openQA tests) so RC code of terraform can be trusted to ensure that all the tests were done well.

Reply:
thx!